--- iris-1.12.0.orig/lib/iris/fileformats/cf.py	2017-01-31 09:45:23.000000000 -0300
+++ iris-1.12.0/lib/iris/fileformats/cf.py	2017-02-11 14:55:52.952101496 -0300
@@ -684,7 +684,10 @@
         # Determine whether we have a string-valued scalar label
         # i.e. a character variable that only has one dimension (the length of the string).
         if self.ndim == 1:
-            data = np.array([''.join(label_data).strip()])
+            label_string = b''.join(label_data).strip()
+            if six.PY3:
+                label_string = label_string.decode('utf8')
+            data = np.array([label_string])
         else:
             # Determine the index of the string dimension.
             str_dim = self.dimensions.index(str_dim_name)
