From 2b8d2a7e43c1fc41e18066a4ea3f10054bd92820 Mon Sep 17 00:00:00 2001
From: ocefpaf <ocefpaf@gmail.com>
Date: Wed, 10 May 2017 09:18:35 -0300
Subject: [PATCH 1/2] pandas >0.20

---
 lib/iris/pandas.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/lib/iris/pandas.py b/lib/iris/pandas.py
index 26ce5c5..80100ec 100644
--- a/lib/iris/pandas.py
+++ b/lib/iris/pandas.py
@@ -1,4 +1,4 @@
-# (C) British Crown Copyright 2013 - 2016, Met Office
+# (C) British Crown Copyright 2013 - 2017, Met Office
 #
 # This file is part of Iris.
 #
@@ -30,7 +30,10 @@ import cf_units
 from cf_units import Unit
 import netcdftime
 import numpy as np
-import pandas
+try:
+    from pandas.core.indexes.datetimes import DatetimeIndex
+except ImportError:
+    from pandas.tseries.index import DatetimeIndex
 
 import iris
 from iris.coords import AuxCoord, DimCoord
@@ -49,7 +52,7 @@ def _add_iris_coord(cube, name, points, dim, calendar=None):
         calendar = cf_units.CALENDAR_GREGORIAN
 
     # Convert pandas datetime objects to python datetime obejcts.
-    if isinstance(points, pandas.tseries.index.DatetimeIndex):
+    if isinstance(points, DatetimeIndex):
         points = np.array([i.to_datetime() for i in points])
 
     # Convert datetime objects to Iris' current datetime representation.
-- 
2.1.4


From 004ef6ff01f0ea950c48cef0a0a288ce3969620d Mon Sep 17 00:00:00 2001
From: ocefpaf <ocefpaf@gmail.com>
Date: Wed, 10 May 2017 09:33:09 -0300
Subject: [PATCH 2/2] add comment and global pandas

---
 lib/iris/pandas.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/iris/pandas.py b/lib/iris/pandas.py
index 80100ec..e347f76 100644
--- a/lib/iris/pandas.py
+++ b/lib/iris/pandas.py
@@ -30,10 +30,11 @@ import cf_units
 from cf_units import Unit
 import netcdftime
 import numpy as np
+import pandas
 try:
-    from pandas.core.indexes.datetimes import DatetimeIndex
+    from pandas.core.indexes.datetimes import DatetimeIndex  # pandas >=0.20
 except ImportError:
-    from pandas.tseries.index import DatetimeIndex
+    from pandas.tseries.index import DatetimeIndex  # pandas <0.20
 
 import iris
 from iris.coords import AuxCoord, DimCoord
-- 
2.1.4

